jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/DefaultRule.java

```diff
@@ -243,11 +246,35 @@ private JExpression getDefaultSet(JType fieldType, JsonNode node) {
 
     }
 
-    private JExpression getDefaultEnum(JType fieldType, JsonNode node) {
+    private JType getParameterType(JDefinedClass fieldType, String methodName){
 
-        JInvocation invokeFromValue = ((JClass) fieldType).staticInvoke("fromValue");
-        invokeFromValue.arg(node.asText());
+        for (JMethod method : fieldType.methods()) {
+            if (EnumRule.FROM_VALUE_METHOD_NAME.equals(method.name())
+                    && (method.mods().getValue() & JMod.PUBLIC) == JMod.PUBLIC
+                    && (method.mods().getValue() & JMod.STATIC) == JMod.STATIC) {
+                final JType[] type = method.listParamTypes();
+                if (type.length != 1)
+                    throw new GenerationException("Factory method '" + EnumRule.FROM_VALUE_METHOD_NAME + "' should has only one parameter in " + fieldType);
+                return type[0];
+            }
+        }
+        throw new GenerationException("Factory method '" + EnumRule.FROM_VALUE_METHOD_NAME + "' is not found in " + fieldType);
+
+    }
 
+    private JExpression getDefaultEnum(JDefinedClass fieldType, JsonNode node) {
+
+        JInvocation invokeFromValue = fieldType.staticInvoke(EnumRule.FROM_VALUE_METHOD_NAME);
+        JType paramType = getParameterType(fieldType, EnumRule.FROM_VALUE_METHOD_NAME).unboxify();
+        if (paramType == paramType.owner().INT) {
```

Comment:
Man, the conversation is beautiful!
I understood your idea, it is reasonable because signatures of both methods  are more or less similar. And right after that I've seen that adding some more if-else into the `getDefaultValue` is a crime :)
Much more clever to use something like `getFactory(JType::type).apply(JsonNode::node)`.
I'll check later how easy to make it :)
Regards!